nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestConvertRecord.java

```diff
@@ -434,7 +435,10 @@ public void testJSONDroppingUnknownFields() throws InitializationException, IOEx
         runner.assertAllFlowFilesTransferred(ConvertRecord.REL_SUCCESS, 1);
 
         final MockFlowFile flowFile = runner.getFlowFilesForRelationship(ConvertRecord.REL_SUCCESS).getFirst();
-        assertFalse(flowFile.getContent().contains("fieldThatShouldBeRemoved"));
+        final List<String> undefinedTopLevelKeys = List.of("undefinedKey", "undefinedObjectArray", "undefinedObject", "undefinedScalarArray");
+
+        undefinedTopLevelKeys.forEach(undefinedTopLevelKey -> assertFalse(flowFile.getContent().contains(undefinedTopLevelKey)));
```

Comment:
Could this be simplified to a single assert that the string "undefined" is not in the result?  Perhaps you could add a comment stating this covers all of the cases "undefinedKey", "undefinedObjectArray", "undefinedObject", "undefinedScalarArray", "undefinedKeyInObject"